content/version_control/VCS.tex

```diff
@@ -0,0 +1,174 @@
+    Branching and merging
+  }
+  \item {   
+    Ability to trace each change made
+  }
+  \item {   
+	Allows multiple developers to work simultaneously
+  }
+  \end{itemize}
+\end{frame}
+
+\begin{frame}{Types of VCS}{Centralized VCS}
+\section{Types of VCS}
+\subsection{Centralized VCS}
+  \begin{columns}
+\column{0.5\textwidth}
+\begin{itemize}
+  \item {
+    Centralized VCS keeps the history of changes on a central server.
+  }
+  \item {
+    Designed with the intent that there is only one true source  
+  }
+  \item {
+    If you want to make a copy of your data, you have to copy/paste it, literally.  
+  }
+  \item{
+	Sourceforge.net uses this type of versioning.  
+  }
+  \item{
+      \textbf{Example:} SVN (Subversion)
+  }  
+  \end{itemize}
+\column{0.5\textwidth}
+\begin{figure}
+	 \includegraphics[width=.9\textwidth]{images/cvcs}
+	 \caption{Centralized Version System \cite{website}}
+ \end{figure}
+\end{columns}
+\end{frame}	
+
+\begin{frame}{Types of VCS}{Decentralized VCS}
+\subsection{Distributed VCS}
+\begin{columns}
+\column{0.5\textwidth}
+\begin{itemize}
+		  \item {
+		    In distributed VCS, everyone has a local copy of the entire work’s history
+		  }
+		  \item {
+		    Each repository is as good as the other
+		  }
+		  \item{
+			Prone to central server crash
+		  }
+		  \item{
+			Your local copy is a repository, and you can commit to it and get all benefits of source control - Offline Source Control.
+		  }	  
+		  \item{
+			Mozilla Firefox uses this kind of versioning.  
+		  }
+		  \item{
+           \textbf{ Example: }Git		  
+		  }
+	  \end{itemize}
+\column{0.5\textwidth}
+\begin{figure}
+	 \includegraphics[width=.9\textwidth]{images/dvcs}
+	 \caption{Distributed Version System \cite{website}}
+ \end{figure}
+\end{columns}
+\end{frame}	
+
+\begin{frame}[allowframebreaks]{Types of VCS}{Git vs SVN}
+\subsection{Git vs SVN}
+
+\textbf{Pros of using SVN}
+\begin{itemize}
+    \item Subversion has better GUIs than Git
+    \item Working through versions in Git is tougher than SVN. Git uses SHA-1 hashes while SVN uses sequential revision numbers. 
+\end{itemize}
+\textbf{Pros of using Git}
+\begin{itemize}
+    \item Git is much faster than SVN as they do not have to keep communicating with a central server when changes are made.
+    \item Very secure. Every file and commit is checksummed.
+    \item The Git branching model is much better than any other VCS'.
+    \item Git repository file formats are simple and hence  repair and corruption is rare. Space requirements are small
+    \item As it is a DVCS, scalability is not an issue. 
+\end{itemize}
+\framebreak
+\begin{figure}
+	\includegraphics[scale=0.45]{images/better-git}
+	\caption{Comparison of Git and SVN speeds\cite{better}}
+\end{figure}
+
+\end{frame}
+
+
+
+\begin{frame}{Take home message}
+\section{Take home message}
+\begin{itemize}
+    \item There is not much difference between the most popular VCSs. It ultimately comes down to the use case.
```

Comment:
Sorry to be a pain in the ass! I think these are the last few changes.

I was on a bit of a rush last time I wrote. What I meant was, that the main point of this slide set should be about *why students/we should use VCS*. I think these bullets are fine, but maybe add a slide before this one with the reasoning behind this, even the following image with explanations will probably suffice:
![file naming](https://doctorfreelance.com/wp-content/uploads/2016/05/file-naming-768x566.png)

Once we accept the merge request, maybe I'll add a few bullet points to explain why git vs svn, hopefully, that will make it clearer

Reply:
I think slide 3 says this. I will add this picture to the slide though

Reply:
And that's fine, but the take home message should also focus on this. So I'd add it at the end